nucleus/slice.py

```diff
@@ -27,3 +29,42 @@ def info(self, id_type: str = DATASET_ITEM_ID_KEY) -> dict:
         }
         """
         return self._client.slice_info(self.slice_id, id_type)
+
+    def download(self, id_type):
```

Comment:
I would put this function in folder references as a helper script. The logic here isn't very obvious and a user may want to customize it. Also I would download not only images but also all the metadata in a separate file as well.